pkg/operator/ceph/cluster/cluster.go

```diff
@@ -183,3 +251,15 @@ func clusterChanged(oldCluster, newCluster cephv1beta1.ClusterSpec) bool {
 
 	return changeFound
 }
+
+func extractCephVersion(version string) (string, error) {
+	switch {
+	case strings.Index(version, cephv1beta1.Luminous) != -1:
```

Comment:
Is there a reason to use `Index` over `Contains`? https://golang.org/pkg/strings/#Contains

Considering the future, do you think this should be made easier to modify? Maybe something like defining supported versions in a list that can be changed with just a 1 or 2 line edit?
```go
supportedVersions := []string{cephv1beta1.Luminous, cephv1beta1.Mimic, cephv1beta1.Nautilus}
for sv := range supportedVersions {
    if strings.Index(version, sv) != -1 {
        return sv, nil
    }
}
return "", fmt.Errorf("failed to parse version from: %s", version)
```

Reply:
Just an old habit to use `Index()`, but i'll change it to `Contains` for readability.

Reply:
yes, i like the `for` loop for readability as well.